lib/scss_lint/plugins/linter_dir.rb

```diff
@@ -19,6 +19,29 @@ def load
       def ruby_files
         Dir.glob(File.expand_path(File.join(@dir, '**', '*.rb')))
       end
+
+      # Returns the {SCSSLint::Config} for this directory.
+      #
+      # This is intended to be merged with the configuration that loaded this
+      # plugin.
+      #
+      # @return [SCSSLint::Config]
+      def plugin_config
+        file = plugin_config_file
+
+        if File.exist?(file)
+          Config.load(file, merge_with_default: false)
+        else
+          Config.new({})
+        end
+      end
+
+      # Path of the configuration file to attempt to load for this directory.
+      #
+      # @return [String]
+      def plugin_config_file
+        File.join(@dir, Config::FILE_NAME)
```

Comment:
To reduce indirection, this line could probably be inlined directly in the `plugin_config` method (since you don't use it elsewhere).